messages/index.js

```diff
@@ -5,79 +5,88 @@ For a complete walkthrough of creating this type of bot see the article at
 
 
 const LuisModelUrl = 'https://' + luisAPIHostName + '/luis/v1/application?id=' + luisAppId + '&subscription-key=' + luisAPIKey;
 
 // Main dialog with LUIS
-var recognizer = new builder.LuisRecognizer(LuisModelUrl);
-var intents = new builder.IntentDialog({ recognizers: [recognizer] })
+const recognizer = new builder.LuisRecognizer(LuisModelUrl);
+const intents = new builder.IntentDialog({ recognizers: [recognizer] })
     .matches('greeting', (session, args) => {
         session.send('hello i am clappy', session.message.text);
     })
     .matches('actorForRole', (session, args) => {
-        var movieTitle = args.entities.filter(function(entity) {
-            return entity.type === 'movieTitle';
-        })[0].entity;
-        var characterName = args.entities.filter(function(entity) {
-            return entity.type === 'characterName';
-        })[0].entity;
+        const movieEntity = args.entities.filter((entity) => entity.type === 'movieTitle');
+        const characterEntity = args.entities.filter((entity) => entity.type === 'characterName');
 
-        var teststring = 'intent: actorForRole, movie title: ' + movieTitle + ', character name: ' + characterName;
-        var movieSearchUrl = movieUrl+'search/movie/?query=' + movieTitle + '&api_key='+ movieKey;
+        const movieTitle = movieEntity.length ? movieEntity[0].entity : null;
+        const characterName = characterEntity.length ? characterEntity[0].entity : null;
 
-        session.send(teststring);
-
-        request(movieSearchUrl, function (error, response, body) {
-        if (!error && response.statusCode == 200) {
+        if (!movieTitle) {
+            return session.send('Sorry, I don\'t understand which movie you are asking about.');
+        } else if (!characterName) {
+            return session.send('Sorry, I don\'t understand which movie character you are asking about.');
+        }
 
-            var movieId = JSON.parse(body).results[0].id;
-            var movieCreditsUrl = movieUrl+'movie/' + movieId + '/credits?api_key='+ movieKey;
+        const teststring = 'intent: actorForRole, movie title: ' + movieTitle + ', character name: ' + characterName;
+        session.send(teststring);
 
-            request(movieCreditsUrl, function (error, response, body) {
-                if (!error && response.statusCode == 200) {
-                    var character = JSON.parse(body).cast.filter(function(c) {
-                        return c.character.toLowerCase().includes(characterName);
-                    })[0];
-                    session.send("I think the person you're looking for is " + character.name);
-                }
-            });
-        }
+        const movieSearchUrl = movieUrl+'search/movie/?query=' + movieTitle + '&api_key='+ movieKey;
+
+        request(movieSearchUrl, (error, response, body) => {
+            if (!error && response.statusCode == 200) {
+
+                const movieId = JSON.parse(body).results[0].id;
```

Comment:
still need to handle if there are no results, and also a thing to handle multiple results